src/backend/base/langflow/components/data/kb_ingest.py

```diff
@@ -466,10 +472,18 @@ def is_valid_collection_name(self, name, min_length: int = 3, max_length: int =
         # Check allowed characters (condition 3)
         return re.match(r"^[a-zA-Z0-9_-]+$", name) is not None
 
+    async def _kb_path(self) -> Path:
+        current_user = await _get_current_user(self.langflow_api_key)
+
+        kb_root = self._get_kb_root()
+        kb_user = current_user.username
+
+        return kb_root / kb_user / self.knowledge_base
```

Comment:
Maybe set an instance variable and check with getattr just so we don't have to do this repeatedly.